appcues/src/main/java/com/appcues/debugger/screencapture/AndroidTargetingStrategy.kt

```diff
@@ -195,16 +210,17 @@ private fun View.extractResourceName(): String? {
     }
 }
 
-private fun SemanticsNode.asCaptureView(context: Context): ViewElement {
+private fun SemanticsNode.asCaptureView(context: Context, screenBounds: Rect): ViewElement? {
     val displayMetrics = context.resources.displayMetrics
     val density = displayMetrics.density
 
-    var childElements: List<ViewElement>? = children.map {
-        it.asCaptureView(context)
+    // if the view is not currently in the screenshot image (scrolled away), ignore
+    if (screenBounds.contains(unclippedGlobalBounds).not()) {
+        return null
     }
 
-    if (childElements?.isEmpty() == true) {
-        childElements = null
+    val childElements: List<ViewElement> = children.mapNotNull {
+        it.asCaptureView(context, screenBounds)
     }
 
     return selector().let {
```

Comment:
wondering if there is a way to make both `isFullyVisible()` in a place that is similar in relation

Reply:
do you mean check the Compose views for isFullyVisible?

Reply:
yes, right now we are only checking that for View class correct?